### PR TITLE
PERF-#4703: Improve performance in accessing `ser.cat.categories`, `ser.cat.ordered`, and `ser.__array_priority__`

### LINKED ISSUE #4703: PERF: avoid to_pandas/default_to_pandas
in `Series.__array_priority__` and CategoricalMethods

### PR DESCRIPTION
**Pull request by jbrockmendel**
Resolves #4703

## What do these changes do?

- [x] commit message follows format outlined [here](https://modin.readthedocs.io/en/latest/development/contributing.html#commit-message-formatting)
- [x] passes `flake8 modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] passes `black --check modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] signed commit with `git commit -s` 
- [x] Resolves #4703
- [ ] tests added and passing
- [ ] module layout described at `docs/development/architecture.rst` is up-to-date 
- [x] added (Issue Number: PR title (PR Number)) and github username to release notes for next major release

### CODE REVIEW

**Review by mvashishtha — APPROVED** [review]
LGTM. Thanks @jbrockmendel

**Comment by mvashishtha** [comment]
@jbrockmendel we require that contributors sign off on commits. could you please follow the instructions [here](https://modin.readthedocs.io/en/stable/development/contributing.html?highlight=contributing#certificate-of-origin) to do that?

File: modin/pandas/series.py [hunk 1]
```diff
@@ -70,6 +70,7 @@ class Series(SeriesCompat, BasePandasDataset):
     """
 
     _pandas_class = pandas.Series
+    __array_priority__ = pandas.Series.__array_priority__
```
Comment (pyrito) [inline modin/pandas/series.py:73]: what is `__array_priority__`?

**Review by pyrito — COMMENTED** [review]
Could you please add changes to the release docs?

File: modin/pandas/series.py (hunk 1 above)
Comment (jbrockmendel) [reply modin/pandas/series.py:73]: it is used by numpy to determine when an op should return NotImplemented, e.g. `some_numpy_array.__add__(some_modin_series)`

**Comment by jbrockmendel** [comment]
> Could you please add changes to the release docs?

Not clear on what you're looking for here.  There shouldn't be any user-facing changes.

**Comment by pyrito** [comment]
@jbrockmendel We've usually maintained a discipline of updating this manually: https://github.com/modin-project/modin/blob/master/docs/release_notes/release_notes-0.16.0.rst . However, we are moving to automate these in the future.

**Comment by RehanSD** [comment]
Hi @jbrockmendel! Thank you so much for this PR - I'm sure it'll improve perf. Would you mind adding a note to the release notes (found here: https://github.com/modin-project/modin/blob/master/docs/release_notes/release_notes-0.16.0.rst)? You'd need to add the PR title + number to the Performance Enhancements section (like so: `* PERF-#4703: avoid to_pandas/default_to_pandas (#4704)`), as well as your GitHub handle to the list of contributors for the release!

**Review by pyrito — APPROVED** [review]

**Review by RehanSD — DISMISSED** [review]

**Review by RehanSD — APPROVED** [review]
LGTM!

**Review by YarShev — APPROVED** [review]
LGTM!

**Pull request merged**